project/tests/com/google/daggerquery/executor/models/QueryTest.java

```diff
@@ -202,6 +204,100 @@ public void testExecutingAllPathsQuery_WithNullBindingGraph_ThrowsNullPointerExc
     query.execute(null);
   }
 
+  // Tests for `SOMEPATH` query
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testParsingSomePathQuery_WithOneStringParameter_ThrowsIllegalArgumentException() {
+    Query query = new Query("somepath", "com.google.cats.Cat");
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testParsingSomePathQuery_WithoutParameters_ThrowsIllegalArgumentException() {
+    Query query = new Query("somepath");
+  }
+
+  @Test(expected = NullPointerException.class)
+  public void testParsingSomePathQuery_WithNullParameters_ThrowsNullPointerException() {
+    Query query = new Query("somepath", /*parameters = */ null);
+  }
+
+  @Test
+  public void testExecutingSomePathQuery_WithCapitalizedName_WhenResultContainsOnePathWithTwoNodes() {
+    String[] parameters = {"com.google.Cat", "com.google.Details"};
+    Query query = new Query("SoMePaTh", parameters);
+
+    List<String> queryExecutionResult = query.execute(makeBindingGraph_WithMultiplePathsBetweenTwoNodes());
+
+    String[] expectedOutput = {"com.google.Cat -> com.google.Details"};
+    assertArrayEquals(expectedOutput, queryExecutionResult.toArray());
+  }
+
+  @Test
+  public void testExecutingSomePathQuery_WhenResultContainsOnePathWithThreeNodes() {
+    String[] parameters = {"com.google.CatsFactory", "com.google.Details"};
+    Query query = new Query("somepath", parameters);
+
+    List<String> queryExecutionResult = query.execute(makeBindingGraph_WithMultiplePathsBetweenTwoNodes());
+
+    String[] expectedOutput = {"com.google.CatsFactory -> com.google.Cat -> com.google.Details"};
+    assertArrayEquals(expectedOutput, queryExecutionResult.toArray());
+  }
+
+  @Test
+  public void testExecutingSomePathQuery_WhenResultContainsMultiplePaths() {
+    String[] parameters = {"com.google.Component", "com.google.Details"};
+    Query query = new Query("somepath", parameters);
+
+    List<String> queryExecutionResult = query.execute(makeBindingGraph_WithMultiplePathsBetweenTwoNodes());
+
+    Set<String> possibleOutputs = Sets.newHashSet(
+        "com.google.Component -> com.google.Details",
+        "com.google.Component -> com.google.Cat -> com.google.Details",
+        "com.google.Component -> com.google.CatsFactory -> com.google.Cat -> com.google.Details",
+        "com.google.Component -> com.google.Helper -> com.google.CatsFactory -> com.google.Cat -> com.google.Details"
+    );
+
+
+    assertEquals(1, queryExecutionResult.size());
+    assertTrue(possibleOutputs.contains(queryExecutionResult.get(0)));
```

Comment:
This was quite a smart decision. 
It makes it clear that you cannot make any assumptions as to which path will be chosen.

Reply:
Yes exactly! 🚀